### PR TITLE
pubsub: Add topics

### PR DESCRIPTION
**Pull request by rakyll**
/cc @broady

### CODE REVIEW

File: pubsub/topics/main.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	// [START create_topic]
+	t, err := c.NewTopic(ctx, topic)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("topic created: %v\n", t.Name())
+	// [END create_topic]
+}
+
+func list(c *pubsub.Client) {
```
Comment (broady) [inline pubsub/topics/main.go]: Return a slice, so it's testable.

File: pubsub/topics/main.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	ctx := context.Background()
+	// [START auth]
+	proj := os.Getenv("GCLOUD_PROJECT")
+	if proj == "" {
+		fmt.Fprintf(os.Stderr, "GCLOUD_PROJECT environment variable must be set.\n")
+		os.Exit(1)
+	}
+	c, err := pubsub.NewClient(ctx, proj)
+	if err != nil {
+		log.Fatal(err)
```
Comment (broady) [inline pubsub/topics/main.go]: `"Could not create pubsub Client: %v", err`

File: pubsub/topics/main.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+)
+
+func main() {
+	ctx := context.Background()
+	// [START auth]
+	proj := os.Getenv("GCLOUD_PROJECT")
+	if proj == "" {
+		fmt.Fprintf(os.Stderr, "GCLOUD_PROJECT environment variable must be set.\n")
+		os.Exit(1)
+	}
```
Comment (broady) [inline pubsub/topics/main.go]: Take project as a parameter?
We should probably standardize here. What do you lean towards?

File: pubsub/topics/main.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+	// List all the topics from the project.
+	list(c)
+
+	const topic = "example-topic"
+	// Create a new topic called example-topic.
+	create(c, topic)
+
+	// Publish a text message on the created topic.
+	publish(c, topic, "hello world!")
```
Comment (broady) [inline pubsub/topics/main.go]: Should this sample also create a subscription so it can pull the message?

File: pubsub/topics/main.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+func main() {
+	ctx := context.Background()
+	// [START auth]
+	proj := os.Getenv("GCLOUD_PROJECT")
+	if proj == "" {
+		fmt.Fprintf(os.Stderr, "GCLOUD_PROJECT environment variable must be set.\n")
+		os.Exit(1)
+	}
+	c, err := pubsub.NewClient(ctx, proj)
```
Comment (broady) [inline pubsub/topics/main.go]: s/c/client/

File: pubsub/topics/main.go (hunk 3 above)
Comment (rakyll) [inline pubsub/topics/main.go]: Like the env var better, I don't have to deal with os.Args.

File: pubsub/topics/main.go (hunk 4 above)
Comment (rakyll) [inline pubsub/topics/main.go]: subscriptions program will do that.

File: pubsub/topics/main.go (hunk 2 above)
Comment (rakyll) [inline pubsub/topics/main.go]: Done

File: pubsub/topics/main.go (hunk 1 above)
Comment (rakyll) [inline pubsub/topics/main.go]: Done

File: pubsub/topics/main.go (hunk 5 above)
Comment (rakyll) [inline pubsub/topics/main.go]: Done

**Comment by rakyll** [comment]
PTAL

**Comment by broady** [comment]
Needs tests before merging. So I suggest returning errors from the functions.

**Comment by rakyll** [comment]
PTAL, tests are done.

**Comment by broady** [comment]
LGTM when green on Travis :)

**Comment by rakyll** [comment]
Travis is taking so long, tested locally.

**Pull request merged**